hs/CalendarResponse.hs

```diff
@@ -0,0 +1,230 @@
+module CalendarResponse where
+
+import Data.List (sort)
+import Data.List.Split (splitOn)
+import Data.Time (Day, UTCTime, addDays, formatTime, getCurrentTime, utctDay)
+import Happstack.Server (ServerPart, Response, toResponse)
+import Control.Monad.IO.Class (liftIO)
+import System.Locale
+import Config (firstMondayFall, lastWednesdayFall ,firstMondayWinter, lastMondayWinter, outDay)
+import Database.CourseQueries (returnTutorialTimes, returnLectureTimes)
+import qualified Data.Text as T
+import Database.Tables as Tables
+
+-- | Returns a CSV file of events as requested by the user.
+calendarResponse :: String -> ServerPart Response
+calendarResponse courses =
+    liftIO $ getCalendar courses
+
+-- | Gets together all the pieces of the program.
+getCalendar :: String -> IO Response
+getCalendar courses = do
+    let courseInfo = getCoursesInfo courses
+    databaseInfo <- sequence $ map pullDatabase courseInfo
+    currentTime <- getCurrentTime
+    let timeSystem = getTime currentTime
+    let events = concat $ map (getEvent timeSystem) databaseInfo
+    return $ toResponse $ getCSV $ events
+
+-- | Generates a properly formatted current date and time.
+getTime :: UTCTime -> String
+getTime currentTime = date ++ "T" ++ hour ++ "Z"
+    where
+    date = formatTime defaultTimeLocale "%Y%m%d" (utctDay currentTime)
+    hour = formatTime defaultTimeLocale "%H%M%S" currentTime
+
+-- | Generates a string representing a CSV file.
+getCSV :: [String] -> String
+getCSV events =  if events == [] then "" else unlines $ header ++ events ++ bottom
+    where
+    header = ["BEGIN:VCALENDAR", "VERSION:2.0", "PRODID:-//Courseography//Calendar", "CALSCALE:GREGORIAN", "METHOD:PUBLISH"]
+    bottom = ["END:VCALENDAR"]
+
+-- ** Cookie Information
+
+-- | Obtains a list with all the information about the courses obtained from the cookies.
+getCoursesInfo :: String -> [(String, String, String)]
+getCoursesInfo courses = map courseInfo allCourses
+    where
+    courseInfo course = (course !! 0, course !! 1, course !! 2)
+    allCourses = map (splitOn "-") byCourse
+    byCourse = splitOn "_" courses
+
+-- ** Database Information
+
+-- | Pulls out the information for each course from the database.
+pullDatabase :: (String, String, String) -> IO ([Time], String, String, String)
+pullDatabase (code, sect, session) =
+    if take 1 sect == "L"
+    then getData $ returnLectureTimes (T.pack code) (T.pack sect) (T.pack session)
+    else getData $ returnTutorialTimes (T.pack code) (T.pack sect) (T.pack session)
+    where
+    getData info = do
+        courseInfo <- info 
+        return (courseInfo, code, session, sect)
+
+-- ** Event Creation
+
+-- | Generates an event for each course.
+getEvent :: String -> ([Time], String, String, String) -> [String]
+getEvent timeSystem (timeFields, code, session, sect) = concat $ eventsByCourse code session sect start end dates timeSystem
```

Comment:
Some of the lines are a little long, and are difficult to read in some editors/github. A few one liners in this file are put onto the next line, it would be good to do that for some of the other ones as well.